dysnix/bsc/templates/scripts/_init_from_gcs.tpl

```diff
@@ -97,45 +98,77 @@ TIMESTAMP_0="$(${S5CMD} cat s3://${TIMESTAMP_URL})"
 
 # we're ready to perform actual data sync
 
-# we're done when both are true
+# we're done when all are true
 # 1) start and stop timestamps did not changed during data sync - no process started or finished updating the cloud
-# 2) 0 objects copied
+# 2) start timestamp is before stop timestamp - no process is in progress updating the cloud
+# 3) 0 objects copied
 SYNC=2
 CLEANUP=1
 while [ "${SYNC}" -gt 0 ] ; do
 
     # Cleanup
     if [ ${CLEANUP} -eq 1 ];then
-      echo "$(date -Iseconds) Cleaning up local dir ..."
-      mkdir -p ${DATA_DIR}/geth
-      mv ${DATA_DIR}/geth ${DATA_DIR}/geth.old && rm -rf ${DATA_DIR}/geth.old &
+      echo "$(date -Iseconds) Cleaning up local dir ${GETH_DIR} ..."
+      mkdir -p "${GETH_DIR}"
+      mv "${GETH_DIR}" "${GETH_DIR}.old" && rm -rf "${GETH_DIR}.old" &
       CLEANUP=0
     fi
 
-    # sync from cloud to local disk, without removing existing [missing in the cloud] files
+    # sync from cloud to local disk, with removing existing [missing in the cloud] files
     # run multiple syncs in background
 
-    # we don't wanna sync ancient data here
-    time ${S5CMD} sync ${EXCLUDE_ANCIENT} s3://${STATE_SRC}/* ${CHAINDATA_DIR}/ > cplist_state.txt &
+    time ${S5CMD} sync --delete s3://${STATE_SRC}/* ${STATE_TMP_DIR}/ > cplist_state.txt &
     STATE_CP_PID=$!
-    time nice ${S5CMD} sync --delete --part-size 200 --concurrency 2 ${EXCLUDE_STATE} s3://${ANCIENT_SRC}/* ${CHAINDATA_DIR}/ancient/ > cplist_ancient.txt &
+    time nice ${S5CMD} sync --delete --part-size 200 --concurrency 2 s3://${ANCIENT_SRC}/* ${ANCIENT_TMP_DIR}/ > cplist_ancient.txt &
```

Comment:
is there any point to keep `--delete` option if we `CLEANUP=1` on every script start?

Reply:
Yes, as we [usually] use full cleanup just once inside the loop, but we may hit into objects become missing in the source (due to source update at this time) during the download process. We don't need unused/extra files on disk, that's the way for it to stay forever and become space consuming eventually. We'll delete all the unneeded local files with `--delete` .